tests/TestData.cmake

```diff
@@ -0,0 +1,109 @@
+)
+mark_as_advanced(GIRDER_EXTERNAL_DATA_ROOT)
+
+# Contains the "data store" path for sharing downloaded resources
+# between build directories.
+set(GIRDER_EXTERNAL_DATA_STORE "${PROJECT_BINARY_DIR}/store"
+  CACHE STRING "The location where downloaded files are cached.  Can be shared between build directories."
+)
+
+# Set module level variables for ExternalData
+set(ExternalData_BINARY_ROOT "${GIRDER_EXTERNAL_DATA_ROOT}")
+set(ExternalData_OBJECT_STORES "${GIRDER_EXTERNAL_DATA_STORE}")
+
+# Set the top level file path where data key files are stored.
+# This has to be external to the project source directory because
+# we need to build the directory tree with key files provided by
+# plugins as well.
+set(ExternalData_SOURCE_ROOT
+  "${_build_data_path}"
+)
+
+# Set up the data source directory by copying the main data source
+# path into the build.
+function(_setup_base_data_path)
+  # we rebuild this directory on every cmake invocation to avoid
+  # stale files and handling of merging directories.  The overhead
+  # should be minimal because it should only contain a series of
+  # very small text files.
+  if(EXISTS "${_build_data_path}")
+    file(REMOVE_RECURSE "${_build_data_path}")
+    file(MAKE_DIRECTORY "${_build_data_path}")
+  endif()
+  file(GLOB _base_data_files "${_source_data_path}/*")
+  file(COPY ${_base_data_files} DESTINATION "${_build_data_path}")
+endfunction()
+
+# Add a plugin data directory to the root data store.
+function(add_plugin_data_path plugin_name plugin_dir)
+  set(_plugin_source_data_path "${plugin_dir}/${_plugin_rel_data_path}")
+  if(EXISTS "${_plugin_source_data_path}")
+    file(MAKE_DIRECTORY "${_plugin_build_data_path}/${plugin_name}")
+    file(GLOB _plugin_data_files "${_plugin_source_data_path}/*")
+    file(COPY ${_plugin_data_files} DESTINATION "${_plugin_build_data_path}/${plugin_name}")
+  endif()
+endfunction()
+
+# Adds a new mirror hosting external data.  The argument passed to this
+# method should be a url template similar to the default mirror:
+# 
+#   https://midas3.kitware.com/midas/api/rest?method=midas.bitstream.download&checksum=%(hash)&algorithm=%(algo)
+# 
+# where "%(algo)" will be replaced by a hashing algorithm such as "md5" or
+# "sha512" and "%(hash)" will be the corresponding hash value.
+#
+# This is provided as a function because in the future we may want to provide
+# a custom fetch script to handle things like authentication.
+function(add_external_data_mirror url_template)
+  list(APPEND ExternalData_URL_TEMPLATES url_template)
+endfunction()
+
+# These are wrappers around ExternalData functions that have access to the
+# locally scoped variables such as the mirror list and the data source
+# root directory.
+#   See https://cmake.org/cmake/help/v3.3/module/ExternalData.html#module-functions
+function(girder_add_test)
+  ExternalData_add_test(${ARGN})
+endfunction()
+
+function(girder_add_target)
```

Comment:
Are these the most appropriate names, since they're in the global CMake namespace? I could see some future functionality that adds a custom target in a more general way.